### ISSUE TITLE
Projects with bind mounts on Windows fails if its path changes

### ISSUE THREAD

**Issue by funkyhippo** [docker/compose#7327]
## Description of the issue

On compose projects that have bind mounts, running `docker-compose up` after the initial build/setup fails if its path changes. 

This does not seem to be an issue under MacOS or Ubuntu (so basically, **only Windows seems to be affected**).

Note that I can mitigate the issue by restarting the Docker daemon.

## Context information (for bug reports)

**Output of `docker-compose version`**
```
docker-compose version 1.25.4, build 8d51620a
docker-py version: 4.1.0
CPython version: 3.7.4
OpenSSL version: OpenSSL 1.1.1c  28 May 2019
```

**Output of `docker version`**
```
Client: Docker Engine - Community
 Version:           19.03.8
 API version:       1.40
 Go version:        go1.12.17
 Git commit:        afacb8b
 Built:             Wed Mar 11 01:23:10 2020
 OS/Arch:           windows/amd64
 Experimental:      false

Server: Docker Engine - Community
 Engine:
  Version:          19.03.8
  API version:      1.40 (minimum version 1.12)
  Go version:       go1.12.17
  Git commit:       afacb8b
  Built:            Wed Mar 11 01:29:16 2020
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          v1.2.13
  GitCommit:        7ad184331fa3e55e52b890ea95e65ba581ae3429
 runc:
  Version:          1.0.0-rc10
  GitCommit:        dc9208a3303feef5b3839f4323d9beb36df0a9dd
 docker-init:
  Version:          0.18.0
  GitCommit:        fec3683
```

**Output of `docker-compose config`**
(Make sure to add the relevant `-f` and other flags)
```
services:
  test:
    build:
      context: C:\docker-compoop
    command: python -u /script.py
    environment:
      PYTHONUNBUFFERED: '1'
    volumes:
    - source: C:\docker-compoop\script.py
      target: /script.py
      type: bind
version: '3.7'
```

## Steps to reproduce the issue

1. Clone [this repo](https://github.com/funkyhippo/docker-compoop)
2. Run `docker-compose up` within the repo directory, then kill it after it successfully starts
3. Rename the repo directory (to anything)
4. Run `docker-compose up` within the repo directory again

### Observed result

The bound files will not be seen by the containers; instead, we'll get a `No such file or directory` error for each file.

Full console output:
```powershell
C:\docker-compoop>docker-compose up
Creating network "docker-compoop_default" with the default driver
Creating docker-compoop_test_1 ... done                                                                                                                      Attaching to docker-compoop_test_1
test_1  | Sleeping..
test_1  | Sleeping..
Gracefully stopping... (press Ctrl+C again to force)
Stopping docker-compoop_test_1 ... done                                                                                                                      
C:\docker-compoop>cd ..

C:\>rename docker-compoop renamed-docker-compoop

C:\>cd renamed-docker-compoop

C:\renamed-docker-compoop>docker-compose up
Creating network "renamed-docker-compoop_default" with the default driver
Building test
Step 1/1 : FROM python:3-alpine
 ---> d5e5ad4a4fc0

Successfully built d5e5ad4a4fc0
Successfully tagged renamed-docker-compoop_test:latest
WARNING: Image for service test was built because it did not already exist. To rebuild this image you must use `docker-compose build` or `docker-compose up --build`.
Creating renamed-docker-compoop_test_1 ... done                                                                                                              Attaching to renamed-docker-compoop_test_1
test_1  | python: can't open file '/script.py': [Errno 2] No such file or directory
renamed-docker-compoop_test_1 exited with code 2
```

### Expected result

The bound files should be seen, allowing the container to run properly.

### Stacktrace / full error message

```
test_1  | python: can't open file '/script.py': [Errno 2] No such file or directory
```

## Additional information

This was observed on Windows 10, Version 1909 (Build 18363.720). I was not able to reproduce this on MacOS Catalina nor Ubuntu 18.04.

`docker-compose` was installed via Docker Desktop for Windows.

**Comment by phawxby** [comment]
Definitely some kind of issue with volumes on Windows between Docker Desktop Community 2.2.0.3 and 2.2.0.4. So that would be a regression between Docker 19.03.5 and 19.03.8.

```
Drush\Sql\SqlException: Unable to find a matching SQL Class. Drush cannot find your database connection details. in                                              [error]
phar:///usr/local/bin/drush/commands/sql/sql.drush.inc:545
Stack trace:
#0 phar:///usr/local/bin/drush/commands/sql/sql.drush.inc(432): drush_sql_get_class()
#1 phar:///usr/local/bin/drush/includes/command.inc(422): drush_sql_drop('@x.local')
#2 phar:///usr/local/bin/drush/includes/command.inc(231): _drush_invoke_hooks(Array, Array)
#3 phar:///usr/local/bin/drush/includes/command.inc(199): drush_command('@x.local')
#4 phar:///usr/local/bin/drush/lib/Drush/Boot/BaseBoot.php(67): drush_dispatch(Array)
#5 phar:///usr/local/bin/drush/includes/preflight.inc(66): Drush\Boot\BaseBoot->bootstrap_and_dispatch()
#6 phar:///usr/local/bin/drush/includes/startup.inc(462): drush_main()
#7 phar:///usr/local/bin/drush/includes/startup.inc(369): drush_run_main(false, '/', 'Phar detected. ...')
#8 phar:///usr/local/bin/drush/drush(114): drush_startup(Array)
#9 /usr/local/bin/drush(10): require('phar:///usr/loc...')
#10 {main}
Command dispatch complete                                                                                                                                       [notice]
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

docker-compose.yml
```
  drush:
    depends_on:
      - db
      - web
    build:
      context: ./docker
      dockerfile: Dockerfile.drush
    tmpfs:
      - /run
    volumes:
      - ./docroot:/app
      - ./docker/drush:/host
      - ./docker/drush/.drush:/root/.drush
      - ./docker/drush/.ssh:/root/.ssh
```

Dockerfile.drush
```
FROM drush/drush:8

# Update modules
RUN apk add --update \
    rsync \
  && rm -rf /var/cache/apk/*

RUN mkdir -p /run/secrets
COPY ./.ssh/id_rsa /run/secrets/id_rsa

RUN chmod -R 0700 /run/secrets
VOLUME /run/secrets
```

**Comment by funkyhippo** [comment]
Your comment actually made me do some additional investigation, and it looks like this issue isn't related to compose as I had originally thought; I ran the wrong sequence of commands when I was testing it through Docker itself, thus coming to the wrong conclusion that it was an issue with compose.

As for my investigation, I tried downgrading to 2.2.0.3 which used Docker Engine 19.03.5. I had the same issue with both compose and building the image manually and running the container. 

2.2.0.0's download was pulled from the site, so I tried 2.1.0.5 and it worked fine! It looks like the new file sharing implementation from 2.2.0.0 has some bugs that haven't been fixed in the versions following, and that's what I'm seeing.

For reference, I did some additional digging and found a similar issue already reported: https://github.com/docker/for-win/issues/5530. It's not exactly the same issue, but at least the issue with file consistency's been acknowledged. 

 In your case, downgrading to 2.1.0.5 might help.
